### PR TITLE
Improves creating ProcessLog entries out of Exceptions

### PR DESCRIPTION
**Pull request by idlira**
Permits to "build" ProcessLog entries out of an existing Exception.
If the Exception is a HandledException, automatically parses the `messageType` and `messageCount` hints if set.

Fixes: OX-7331

### CODE REVIEW

File: src/main/java/sirius/biz/process/logs/ProcessLog.java [hunk 1]
```diff
@@ -393,6 +395,31 @@ public ProcessLog withFormattedMessage(String message, Object... parameters) {
...
+     * If the provided exception is a {@link HandledException}, also considers the filled
+     * {@link ProcessContext#HINT_MESSAGE_TYPE} to categorize the log in message types and
+     * {@link ProcessContext#HINT_MESSAGE_COUNT} to limit the amount of allowed messages per type.
+     *
+     * @param exception the {@link Exception} to retrieve the message and eventually hints from
+     * @return the log entry itself for fluent method calls
+     */
+    public ProcessLog withException(Exception exception) {
+        this.withMessage(exception.getMessage());
+        if (exception instanceof HandledException) {
```
Comment (sabieber) [inline src/main/java/sirius/biz/process/logs/ProcessLog.java:410]: Optional: Use instanceof pattern matching to remove the casts below like this: `if (exception instanceof HandledException handledException`

File: src/main/java/sirius/biz/process/ErrorContext.java [hunk 2]
```diff
@@ -147,6 +147,7 @@ public String getContextAsString() {
     private void logException(UnaryOperator<String> failureDescription, HandledException exception) {
         if (TaskContext.get().getAdapter() instanceof ProcessContext) {
```
Comment (sabieber) [inline src/main/java/sirius/biz/process/ErrorContext.java:148]: Optional: bind getAdapter to a variable and use instanceof pattern matching to remove cast below

**Review by sabieber — CHANGES_REQUESTED** [review]

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/biz/process/logs/ProcessLog.java [hunk 3]
```diff
@@ -393,6 +395,31 @@ public ProcessLog withFormattedMessage(String message, Object... parameters) {
...
+     * Specifies an exception from where to inherit the message.
+     * <p>
+     * If the provided exception is a {@link HandledException}, also considers the filled
+     * {@link ProcessContext#HINT_MESSAGE_TYPE} to categorize the log in message types and
+     * {@link ProcessContext#HINT_MESSAGE_COUNT} to limit the amount of allowed messages per type.
+     *
+     * @param exception the {@link Exception} to retrieve the message and eventually hints from
+     * @return the log entry itself for fluent method calls
+     */
+    public ProcessLog withException(Exception exception) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/logs/ProcessLog.java:408]: Either limit this to HandledException or move the Exceptions.handle(...) from ProcessEnvironment to here...

**Review by andyHa — CHANGES_REQUESTED** [review]
...

**Review by sabieber — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Pull request merged**